### PR TITLE
fix two issues with COLLECT COUNT

### PR DESCRIPTION
**Pull request by mpoeter**
Copy of #13350 (Jenkins doesn't like branches with emojis).

### Scope & Purpose

Fix two issues with COLLECT COUNT:

* When using a query such as

      FOR i IN 1..1000
        COLLECT AGGREGATE c = COUNT(1)
        RETURN c

  the input variable for COUNT (value `1`) is not needed.
  This PR changes the aggregate variable info in the CollectNode so
  that it forgets about its input variable, allowing further
  optimizer steps to remove the input variable altogether if not
  used elsewhere in the query.

  The above query will now be optimized to

      FOR i IN 1..1000
        COLLECT AGGREGATE c = COUNT()
        RETURN c

* The explainer was not prepared to handle aggregates
  that did not have any input variables, e.g. `AGGREAGTE c = COUNT()`.
  When using them, the explainer produced a JavaScript error.
  This is now fixed.

No CHANGELOG entry added yet because this is a fix/optimization for devel-only functionality that was only very recently added.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

### Testing & Verification

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13533/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13569/

**Comment by mpoeter** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13580/

**Comment by mpoeter** [comment]
Retry: http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13585/

File: tests/js/server/aql/aql-optimizer-collect-count.js [hunk 1]
```diff
@@ -101,11 +102,14 @@ function optimizerCountTestSuite () {
...
         assertEqual("cnt", clusterCollectNode.aggregates[0].outVariable.name);
         assertEqual(clusterCollectNode.aggregates[0].inVariable.name, collectNode.aggregates[0].outVariable.name);
+        assertTrue(/\COLLECT AGGREGATE #[0-9] = LENGTH\(\)/.test(output));
+        assertTrue(/\COLLECT AGGREGATE cnt = SUM\(#[0-9]\)/.test(output));
       } else {
         assertEqual("count", collectNode.collectOptions.method);
         assertEqual(1, collectNode.aggregates.length);
         assertEqual("cnt", collectNode.aggregates[0].outVariable.name);
         assertEqual("LENGTH", collectNode.aggregates[0].type);
+        assertTrue(/\COLLECT AGGREGATE cnt = LENGTH\(\)/.test(output));
```
Comment (jsteemann) [inline tests/js/server/aql/aql-optimizer-collect-count.js:112]:
```suggestion
        assertTrue(/COLLECT AGGREGATE cnt = LENGTH\(\)/.test(output));
```

File: tests/js/server/aql/aql-optimizer-collect-count.js [hunk 2]
```diff
@@ -101,11 +102,14 @@ function optimizerCountTestSuite () {
         assertEqual("SUM", clusterCollectNode.aggregates[0].type);
         assertEqual("cnt", clusterCollectNode.aggregates[0].outVariable.name);
         assertEqual(clusterCollectNode.aggregates[0].inVariable.name, collectNode.aggregates[0].outVariable.name);
+        assertTrue(/\COLLECT AGGREGATE #[0-9] = LENGTH\(\)/.test(output));
+        assertTrue(/\COLLECT AGGREGATE cnt = SUM\(#[0-9]\)/.test(output));
```
Comment (jsteemann) [inline tests/js/server/aql/aql-optimizer-collect-count.js:106]:
```suggestion
        assertTrue(/COLLECT AGGREGATE #[0-9] = LENGTH\(\)/.test(output));
        assertTrue(/COLLECT AGGREGATE cnt = SUM\(#[0-9]\)/.test(output));
```

**Pull request merged**